js/src/components/modals/PostSettingsModal.js

```diff
@@ -2,6 +2,107 @@ import { Button } from "@yoast/components/src/button/Button";
 import { __, sprintf } from "@wordpress/i18n";
 import { useState, useCallback } from "@wordpress/element";
 import Modal from "./Modal";
+import Collapsible from "../SidebarCollapsible";
+import SnippetEditor from "../../containers/SnippetEditor";
+import FacebookContainer from "../../containers/FacebookEditor";
+import TwitterContainer from "../../containers/TwitterEditor";
+import SchemaTabContainer from "../../containers/SchemaTab";
+import AdvancedSettings from "../AdvancedSettings";
+import PropTypes from "prop-types";
+
+/**
+ * Returns the ModalContent.
+ *
+ * We need this in a function since the scripData is not available earlier.
+ *
+ * @param {object} settings An object with settings from the store.
+ *
+ * @returns {array} An array of objects that can be used to render the PostSettingsModal.
+ */
+const modalContent = ( settings ) => [
+	{
+		title: __( "Google preview", "wordpress-seo" ),
+		content: <SnippetEditor hasPaperStyle={ false } />,
+		shouldRender: true,
+	},
+	{
+		title: __( "Facebook preview", "wordpress-seo" ),
+		content: <FacebookContainer />,
+		shouldRender: window.wpseoScriptData.metabox.showSocial.facebook,
+	},
+	{
+		title: __( "Twitter preview", "wordpress-seo" ),
+		content: <TwitterContainer />,
+		shouldRender: window.wpseoScriptData.metabox.showSocial.twitter,
```

Comment:
We don't want to access the script data in our components. We should put them in the store and use selectors.